src/abstractdataframe/abstractdataframe.jl

```diff
@@ -1650,22 +1662,41 @@ julia> flatten(df2, :b)
 │ 3   │ 2     │ r      │
 │ 4   │ 2     │ s      │
 
+julia> df3 = DataFrame(a = [1, 2], b = [[1, 2], [3, 4]], c = [[5, 6], [7]])
+2×3 DataFrame
+│ Row │ a     │ b      │ c      │
+│     │ Int64 │ Array… │ Array… │
+├─────┼───────┼────────┼────────┤
+│ 1   │ 1     │ [1, 2] │ [5, 6] │
+│ 2   │ 2     │ [3, 4] │ [7]    │
+
+julia> flatten(df3, [:b, :c])
+ERROR: ArgumentError: Vector lengths across columns in col, within the same row, must be the same
 ```
 """
-function flatten(df::AbstractDataFrame, col::ColumnIndex)
-    col_to_flatten = df[!, col]
-    lengths = length.(col_to_flatten)
-    new_df = similar(df[!, Not(col)], sum(lengths))
+function flatten(df::AbstractDataFrame,
+                 cols::Union{ColumnIndex, AbstractVector, Regex, Not, Between, All, Colon})
+    idxcols = index(df)[cols]
+    col1 = idxcols isa AbstractVector ? idxcols[1] : idxcols
+    lengths = length.(df[!, col1])
+    for col in idxcols
+        if length.(df[!, col]) != length.(df[!, col1])
+            throw(ArgumentError("Vector lengths across columns in col, within the same row, must be the same"))
+        end
+    end
 
+    new_df = similar(df[!, Not(cols)], sum(lengths))
     for name in _names(new_df)
         repeat_lengths!(new_df[!, name], df[!, name], lengths)
     end
+    for col in idxcols
```

Comment:
before this line add:
```
length(idxcols) > 1 && sort!(idxcols)
```
otherwise the result will be incorrect if the user passed columns in the order that is not increasing as in the original data frame.

Also could you please add a test checking this specific case? Thank you.